ValveResourceFormat/Resource/ResourceTypes/KV3.cs

```
@@ -0,0 +1,592 @@
+                
+                stringBuilder.Append(Environment.NewLine);
+            }
+
+            for (var i = 0; i < indent-1; i++)
+            {
+                stringBuilder.Append("\t");
+            }
+            stringBuilder.Append("}");
+        }
+
+        //Print a value in the correct representation
+        protected void PrintValue(StringBuilder stringBuilder, Type type, Object value, int indent) {
+            if (type == typeof(KVObject))
+            {
+                ((KVObject)value).Serialize(stringBuilder, indent + 1);
+            }
+            else if (type == typeof(FlaggedString))
+            {
+                stringBuilder.Append((String)value);
+            }
+            else if (type == typeof(String))
+            {
+                String str = (String)value;
+
+                if (str.Contains("\n"))
+                {
+                    stringBuilder.Append("\"\"\"");
+                    stringBuilder.Append(str);
+                    stringBuilder.Append("\"\"\"");
+                }
+                else
+                {
+                    stringBuilder.Append("\"");
+                    stringBuilder.Append(str);
+                    stringBuilder.Append("\"");
+                }
+            }
+            else if (type == typeof(Boolean))
+            {
+                if ((Boolean)value)
+                {
+                    stringBuilder.Append("true");
+                }
+                else
+                {
+                    stringBuilder.Append("false");
+                }
+            }
+            else if (type == typeof(Double))
+            {
+                stringBuilder.Append(((Double)value).ToString("#0.000000",CultureInfo.InvariantCulture));
+            }
+            else if (type == typeof(Int32))
+            {
+                stringBuilder.Append((Int32)value);
+            }
+            else
+            {
+                //This shouldn't happen
```

So do something about it, rather than passively swallowing known error cases.